lib/services/local/plugins.go

```diff
@@ -104,25 +107,56 @@ func (s *PluginsService) GetPlugin(ctx context.Context, name string, withSecrets
 
 // GetPlugins implements services.Plugins
 func (s *PluginsService) GetPlugins(ctx context.Context, withSecrets bool) ([]types.Plugin, error) {
-	startKey := backend.Key(pluginsPrefix)
-	result, err := s.backend.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	const pageSize = 1000
+	var results []types.Plugin
+
+	var page []types.Plugin
+	var startKey string
+	var err error
+	for {
+		page, startKey, err = s.ListPlugins(ctx, pageSize, startKey, withSecrets)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		results = append(results, page...)
+		if startKey == "" {
+			break
+		}
+	}
+
+	return results, nil
+}
+
+// ListPlugins returns a paginated list of plugin instances.
+// StartKey is a resource name, which is the suffix of its key.
+func (s *PluginsService) ListPlugins(ctx context.Context, limit int, startKey string, withSecrets bool) ([]types.Plugin, string, error) {
+	startKeyBytes := []byte(startKey)
```

Comment:
Relying on the user to specify the correct prefix in the `startKey` can cause confusion at best, security issues at worst. The `startKey` should just be the resource name and here we can just `backend.Key(pluginsPrefix, startKey)`.